Client/Extensions/UIImageViewExtensions.swift

```diff
@@ -38,21 +38,28 @@ public extension UIImageView {
         }
 
         backgroundColor = nil
-        sd_setImage(with: nil) // cancels any pending SDWebImage operations.
+        let defaults = fallbackFavicon(forUrl: website)
 
         if let url = website, let bundledIcon = FaviconFetcher.getBundledIcon(forUrl: url) {
             self.image = UIImage(contentsOfFile: bundledIcon.filePath)
             finish(bgColor: bundledIcon.bgcolor)
-        } else {
-            let imageURL = URL(string: icon?.url ?? "")
-            let defaults = fallbackFavicon(forUrl: website)
-            self.sd_setImage(with: imageURL, placeholderImage: defaults.image, options: []) {(img, err, _, _) in
-                guard err == nil else {
-                    finish(bgColor: defaults.color)
+        } else if let imageURL = URL(string: icon?.url ?? "") {
+            // TODO: Wrap this part of KF under our umbrella image loading handler
```

Comment:
Should we create a ticket for this TODO or it will be covered in Part2

Reply:
I already wrapped it so will remove the comment :)